File: C/range_sum_query_immutable.c
typedef struct {
    int* histArray;
} NumArray;


NumArray* numArrayCreate(int* nums, int numsSize) {
    NumArray* obj = malloc(sizeof(NumArray));
    obj->histArray = nums;
    
    for(int i = 1; i < numsSize; i++)
        obj->histArray[i] = obj->histArray[i] + obj->histArray[i-1];
    
    return obj;
}

int numArraySumRange(NumArray* obj, int i, int j) {
    if(i == 0) return obj->histArray[j];
        return obj->histArray[j] - obj->histArray[i-1];
}

void numArrayFree(NumArray* obj) {
    
}

/**
 * Your NumArray struct will be instantiated and called as such:
 * NumArray* obj = numArrayCreate(nums, numsSize);
 * int param_1 = numArraySumRange(obj, i, j);
 
 * numArrayFree(obj);
*/

File: Python3/range_sum_query_immutable.py
class NumArray:

    def __init__(self, nums: List[int]):
        if not nums: return
        
        self.histArray = nums
        for i in range(1, len(nums)):
            self.histArray[i] += self.histArray[i-1] 

    def sumRange(self, i: int, j: int) -> int:
        if i == 0: return self.histArray[j]
        return self.histArray[j] - self.histArray[i-1]


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(i,j)
